sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -0,0 +1,1045 @@
+ * Beam Coder. Rather than write a new coder, you can provide your own translation method as
+ * follows:</p>
+ * <pre>{@code
+ * SimpleFunction<Row, String> cassandraOutputValueType = SimpleFunction<Row, String>(){
+ *    public String apply(Row row) {
+ *      return row.getString('myColName');
+ *    }
+ * };
+ * }</pre>
+ *
+ * <h3>Read data from Elasticsearch using HadoopInputFormatIO transform</h3>
+ *
+ * <pre>{@code
+ * Configuration elasticSearchConf = new Configuration();
+ * elasticSearchConf.set("es.nodes", ElasticsearchHostIp);
+ * elasticSearchConf.set("es.port", "9200");
+ * elasticSearchConf.set("es.resource", "ElasticIndexName/ElasticTypeName");
+ * elasticSearchConf.setClass("key.class", {@link org.apache.hadoop.io.Text Text.class}, Object.class);
+ * elasticSearchConf.setClass("value.class", {@link org.elasticsearch.hadoop.mr.LinkedMapWritable LinkedMapWritable.class}, Object.class);
+ * elasticSearchConf.setClass("mapreduce.job.inputformat.class", {@link org.elasticsearch.hadoop.mr.EsInputFormat EsInputFormat.class}, InputFormat.class);
+ * }</pre>
+ *
+ * <p>Call Read transform as follows:
+ * <pre>{@code
+ * PCollection<KV<Text, LinkedMapWritable>> elasticData =
+ *       p.apply("read", HadoopInputFormatIO.<Text, LinkedMapWritable>.read()
+ *                       .withConfiguration(elasticSearchConf));
+ * }
+ * </pre>
+ * Note: The {@link org.elasticsearch.hadoop.mr.EsInputFormat EsInputFormat} key class is {@link org.apache.hadoop.io.Text Text} and value class is {@link org.elasticsearch.hadoop.mr.LinkedMapWritable LinkedMapWritable}.
+ * Both key and value classes have Beam Coders, which means they don't need a translation method specified..
+ */
+
+public class HadoopInputFormatIO {
+  private static final Logger LOG = LoggerFactory.getLogger(HadoopInputFormatIO.class);
+
+  /**
+   * Creates an uninitialized HadoopInputFormatIO.Read. Before use, the {@code Read} must be
+   * initialized with a HadoopInputFormatIO.Read#withConfiguration(HadoopConfiguration) that
+   * specifies the source. A key/value translation may also optionally be specified using
+   * HadoopInputFormatIO.Read#withKeyTranslation/HadoopInputFormatIO.Read#withValueTranslation.
+   */
+  public static <K, V> Read<K, V> read() {
+    return new AutoValue_HadoopInputFormatIO_Read.Builder<K, V>().build();
+  }
+
+  /**
+   * A {@link PTransform} that reads from any data source which implements Hadoop InputFormat.
+   * For e.g. Cassandra, Elasticsearch, HBase, Redis, Postgres, etc. See the class-level Javadoc on
+   * {@link HadoopInputFormatIO} for more information.
+   *
+   * @param <K> Type of keys to be read.
+   * @param <V> Type of values to be read.
+   * @see HadoopInputFormatIO
+   */
+  @AutoValue
+  public abstract static class Read<K, V> extends PTransform<PBegin, PCollection<KV<K, V>>> {
+
+    public static TypeDescriptor<?> inputFormatClass;
```

Comment:
Okay. We will add those variables in existing abstract Builder class using AutoValue.